Dockerfile

```diff
@@ -0,0 +1,434 @@
+RUN ${BUILD_DIR}/scripts/generate_datafed.sh
+RUN cmake -S. -B build			\
+	-DBUILD_REPO_SERVER=False	\
+	-DBUILD_AUTHZ=False		\
+	-DBUILD_CORE_SERVER=False	\
+	-DBUILD_WEB_SERVER=True		\
+	-DBUILD_DOCS=False		\
+	-DBUILD_PYTHON_CLIENT=False	\
+	-DBUILD_FOXX=False		\
+	-DBUILD_COMMON=False
+RUN cmake --build build
+
+ENV NVM_DIR="$NVM_DIR"
+ENV NVM_INC="$NVM_INC"
+ENV NVM_BIN="$NVM_BIN"
+ENV PATH="$NVM_BIN:$PATH"
+
+RUN cmake --build build --target install
+
+FROM ubuntu:focal AS runtime
+
+ARG DATAFED_DIR
+ARG DATAFED_INSTALL_PATH
+ARG BUILD_DIR
+
+# Create datafed user, prefer more secure login options than password
+# Recommended to mount ssh public key on run
+RUN adduser --disabled-password --gecos "" datafed
+
+RUN mkdir -p ${DATAFED_DIR}
+RUN mkdir -p /opt/datafed
+RUN mkdir -p /var/log/datafed
+RUN chown -R datafed:datafed /opt/datafed
+RUN chown -R datafed:datafed /var/log/datafed
+RUN chown -R datafed:datafed ${DATAFED_DIR}
+WORKDIR ${DATAFED_DIR}
+
+RUN apt update
+RUN apt install -y grep libcurl4
+# ENV LD_LIBRARY_PATH="$LD_LIBRARY_PATH:/usr/lib:/usr/local/lib"
+
+FROM runtime AS core
+
+ARG DATAFED_DIR
+ARG DATAFED_INSTALL_PATH
+ARG BUILD_DIR
+ARG LIB_DIR
+
+# The above should also be available at runtime
+ENV DATAFED_INSTALL_PATH="$DATAFED_INSTALL_PATH"
+ENV          DATAFED_DIR="$DATAFED_DIR"
+ENV            BUILD_DIR="$BUILD_DIR"
+
+# copy necessary shared libraries
+# libprotobuf
+COPY --from=dependencies ${LIB_DIR}/libprotobuf.so.3.17.3.0 ${LIB_DIR}/libprotobuf.so.3.17.3.0
+RUN ln -s ${LIB_DIR}/libprotobuf.so.3.17.3.0 ${LIB_DIR}/libprotobuf.so
+
+# libzmq
+COPY --from=dependencies ${LIB_DIR}/libzmq.so.5.2.4 ${LIB_DIR}/libzmq.so.5.2.4
+RUN ln -s ${LIB_DIR}/libzmq.so.5.2.4 ${LIB_DIR}/libzmq.so.5
+RUN ln -s ${LIB_DIR}/libzmq.so.5     ${LIB_DIR}/libzmq.so
+
+# libsodium
+COPY --from=dependencies ${LIB_DIR}/libsodium.so.23.3.0 ${LIB_DIR}/libsodium.so.23.3.0
+RUN ln -s ${LIB_DIR}/libsodium.so.23.3.0 ${LIB_DIR}/libsodium.so.23
+RUN ln -s ${LIB_DIR}/libsodium.so.23     ${LIB_DIR}/libsodium.so
+
+# libboost-program-options
+COPY --from=dependencies /lib/x86_64-linux-gnu/libboost_program_options.so.1.71.0 ${LIB_DIR}/libboost_program_options.so.1.71.0
+RUN ln -s ${LIB_DIR}/libboost_program_options.so.1.71.0 ${LIB_DIR}/libboost_program_options.so
+
+RUN ldconfig
+
+USER datafed
+
+COPY --chown=datafed:datafed ./scripts/generate_datafed.sh     ${DATAFED_DIR}/scripts/generate_datafed.sh
+COPY --chown=datafed:datafed ./scripts/generate_core_config.sh ${DATAFED_DIR}/scripts/generate_core_config.sh
+COPY --chown=datafed:datafed ./scripts/install_core.sh         ${DATAFED_DIR}/scripts/install_core.sh
+COPY --chown=datafed:datafed ./cmake/Version.cmake             ${DATAFED_DIR}/cmake/Version.cmake
+COPY --from=core-build --chown=datafed:datafed ${BUILD_DIR}/core/docker/entrypoint.sh    ${BUILD_DIR}/core/entrypoint.sh
+COPY --from=core-build --chown=datafed:datafed ${DATAFED_INSTALL_PATH}/core/datafed-core ${DATAFED_INSTALL_PATH}/core/datafed-core
```

Comment:
So all of these need to be changed to using the root group. Also you will want to use an env variable or Docker ARG for the user name.